Sources/PackageModel/Manifest.swift

```diff
@@ -165,21 +165,21 @@ public final class Manifest: ObjectIdentifierProtocol {
             return dependencies
         }
         #else
-        guard toolsVersion >= .v5_2 && productFilter != .everything else {
+        guard toolsVersion >= .v5_2 && packageKind != .root else {
             return dependencies
         }
-
-        var requiredDependencies: Set<PackageDependencyDescription> = []
-
+        
+        var requiredDependencyURLs: Set<String> = []
+        
         for target in targetsRequired(for: products) {
             for targetDependency in target.dependencies {
                 if let dependency = packageDependency(referencedBy: targetDependency) {
-                    requiredDependencies.insert(dependency)
+                    requiredDependencyURLs.insert(dependency.url)
                 }
             }
         }
-
-        return Array(requiredDependencies)
+        
```

Comment:
Can we please remove the added invisibles at line 171, 173 and 181?

Reply:
We can surely remove them in main, but I don't think another multi-hour CI run is necessary for this to do it in 5.4 as well.